### PR TITLE
Remove `extensions.*` and `disabled_roles` settings

### PR DESCRIPTION
**Pull request by JuanGarriuz**
### Description

This pull request removes the `extensions.*` and `disabled_roles` because these settings will be deprecated when moving to different registered applications.

Changes:
- Remove the visibility of modules:
  - Remove the view to configure the modules in the UI
  - Remove settings related to configuring the visibility of modules on the configuration file, ability to configure from UI
  - Remove the persistence of visibility of modules on `wazuh-registry.json` file
  - Add a migration to remove the  `extensions` key in the host data stored in `wazuh-registry.json`
- Remove the setting `disabled_roles` from the configuration file and UI.
  - Remove frontend request to check if the `wazuh` application should be visible
  - Remove endpoint to check if the `wazuh` application should be visible
- Remove the logic related to extensions
- Remove managementReducers key from Redux store and related action
  creators and related usage
- Remove toolsReducers key from Redux store and related action
  creators and related usage
- Remove securityReducers key from Redux store and related action
  creators and related usage
- Fix some breadcrumbs
 
### Issues Resolved

https://github.com/wazuh/wazuh-dashboard-plugins/issues/5965

### Evidence

image

### Test

Legend:
:black_circle:: none
:green_circle:: pass
:yellow_circle:: warning
:red_circle:: fail
:white_circle:: not applicable

## UI

| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
| The side menu should not display access to the Modules application | :black_circle: | :black_circle: | :black_circle: |
| Upgrading the application, should remove the extensions key from the wazuh-registry.json | :black_circle: | :black_circle: | :black_circle: |
| The extensions.* and disabled_roles settings should not be displayed in the Wazuh plugin settings application | :black_circle: | :black_circle: | :black_circle: |

**Details**
<details>
<summary>:black_circle: The side menu should not display access to the Modules application</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Upgrading the application, should remove the extensions key from the wazuh-registry.json</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: The extensions.* and disabled_roles settings should not be displayed in the Wazuh plugin settings application</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

## Backend

| Test | Result |
| --- |  --- |
| In a fresh installation, the generated configuration file (wazuh.yml) should not contain reference to `extensions.*` and `disabled_roles` settings. | :black_circle: |

**Details**
<details>
<summary>:black_circle: In a fresh installation, the generated configuration file (wazuh.yml) should not contain reference to `extensions.*` and `disabled_roles` settings.</summary>

</details>

### Check List
- [ ] All tests pass
  - [ ] `yarn test:jest`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Comment by yenienserrano** [comment]
### Test

Legend:
:black_circle:: none
:green_circle:: pass
:yellow_circle:: warning
:red_circle:: fail
:white_circle:: not applicable

## UI

| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
| The side menu should not display access to the Modules application | :black_circle: | :black_circle: | :black_circle: |
| Upgrading the application, should remove the extensions key from the wazuh-registry.json | :black_circle: | :black_circle: | :black_circle: |
| The extensions.* and disabled_roles settings should not be displayed in the Wazuh plugin settings application | :black_circle: | :black_circle: | :black_circle: |

**Details**
<details>
<summary>:black_circle: The side menu should not display access to the Modules application</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Upgrading the application, should remove the extensions key from the wazuh-registry.json</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: The extensions.* and disabled_roles settings should not be displayed in the Wazuh plugin settings application</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

## Backend

| Test | Result |
| --- |  --- |
| In a fresh installation, the generated configuration file (wazuh.yml) should not contain reference to `extensions.*` and `disabled_roles` settings. | :black_circle: |

**Details**
<details>
<summary>:black_circle: In a fresh installation, the generated configuration file (wazuh.yml) should not contain reference to `extensions.*` and `disabled_roles` settings.</summary>

</details>

File: plugins/main/server/lib/generate-alerts/sample-data/osquery.js
Comment (Desvelao) [inline plugins/main/server/lib/generate-alerts/sample-data/osquery.js:1]: **nitpick**: undo the change.

File: scripts/wazuh-alerts-generator/lib/modules/osquery.js
Comment (Desvelao) [inline scripts/wazuh-alerts-generator/lib/modules/osquery.js:1]: **nitpick**: undo the change.

Comment (Desvelao) [reply scripts/wazuh-alerts-generator/lib/modules/osquery.js:1]: Solved in https://github.com/wazuh/wazuh-dashboard-plugins/pull/5988/commits/95bd902d53da7a3546398cce3a0d27e90ffd12bd.

File: plugins/main/server/lib/generate-alerts/sample-data/osquery.js
Comment (Desvelao) [reply plugins/main/server/lib/generate-alerts/sample-data/osquery.js:1]: Solved in https://github.com/wazuh/wazuh-dashboard-plugins/pull/5988/commits/95bd902d53da7a3546398cce3a0d27e90ffd12bd.

**Comment by Desvelao** [comment]
Legend:
:black_circle:: none
:green_circle:: pass
:yellow_circle:: warning
:red_circle:: fail
:white_circle:: not applicable

## UI

| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
| The side menu should not display access to the Modules application | :green_circle: | :black_circle: | :black_circle: |
| Upgrading the application, should remove the extensions key from the wazuh-registry.json | :green_circle: | :black_circle: | :black_circle: |
| The extensions.* and disabled_roles settings should not be displayed in the Wazuh plugin settings application | :green_circle: | :black_circle: | :black_circle: |

**Details**
<details>
<summary>:black_circle: The side menu should not display access to the Modules application</summary>

Chrome - :green_circle:
image

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:green_circle: Upgrading the application, should remove the extensions key from the wazuh-registry.json</summary>

Chrome - :green_circle:
```
{"name":"Wazuh dashboard","app-version":"4.7.0","revision":"00","installationDate":"2023-10-20T09:27:40.686Z","lastRestart":"2023-10-20T09:27:40.686Z","hosts":{"imposter":{"cluster_info":{"manager":"wazuh-manager-master-0","node":"master-node","cluster":"wazuh","status":"enabled","allow_run_as":2},"extensions":{"github":true}}}}
```

```
node@osd:~/kbn$ cat data/wazuh/config/wazuh-registry.json 
{"name":"Wazuh dashboard","app-version":"4.8.0","revision":"00","installationDate":"2023-10-20T13:54:57.200Z","lastRestart":"2023-10-20T13:54:57.200Z","hosts":{"imposter":{"cluster_info":{"manager":"wazuh-manager-master-0","node":"master-node","cluster":"wazuh","status":"enabled","allow_run_as":2}}}}
```

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:green_circle: The extensions.* and disabled_roles settings should not be displayed in the Wazuh plugin settings application</summary>

Chrome - :green_circle:
image

Firefox - :black_circle:

Safari - :black_circle:

</details>

## Backend

| Test | Result |
| --- |  --- |
| In a fresh installation, the generated configuration file (wazuh.yml) should not contain reference to `extensions.*` and `disabled_roles` settings. | :black_circle: |

**Details**
<details>
<summary>:black_circle: In a fresh installation, the generated configuration file (wazuh.yml) should not contain reference to `extensions.*` and `disabled_roles` settings.</summary>

</details>

**Review by Desvelao — APPROVED** [review]
LGTM!

**Review by yenienserrano — APPROVED** [review]
LGTM
